### PR TITLE
adding serial number

### PR DESCRIPTION
**Pull request by d3v1c3nv11**
Added Manifacturer string as  MICROPY_HW_USB_STR_MANUF, use board name as Product string, serial number from mp_hal_get_unique_id

### CODE REVIEW

**Comment by robert-hh** [comment]
That is surely a good contribution. So this is an update of your previous PR. You do not have to open a new PR if you change the old one. You can just push the changed files to the previous branch, and github will update the PR accordingly.
Here: The commit message has the wrong format, and It looks as if the code formatting is wrong as well. You can use uncrustify to force the proper code formatting.

**Comment by d3v1c3nv11** [comment]
> 

sorry, new to this, learning on the fly

**Comment by robert-hh** [comment]
No problem. It took quite a while for me to get a little bit familiar with git. And still I cross my fingers (and make a backup copy) before major changes to the repository, espcially rebase -i.

**Comment by alphaFred** [comment]
@d3v1c3nv11  just a Short question - will the serial number change every time you restart the board?

**Comment by robert-hh** [comment]
Yes & no. It is the unique number which is also used for MAC and returned by machine.unique_id(). But there is a problem in that the unique_id() returns 8 byte, resulting in 16 hex characters, but DESC_STR_MAX is 20. And the way the code is made, the last characters are not initialiazed.

File: ports/mimxrt/tusb_port.c [hunk 1]
```diff
@@ -104,9 +109,31 @@ const uint16_t *tud_descriptor_string_cb(uint8_t index, uint16_t langid) {
...
+                    if (strs[j * 2 + k] > 0x39) {
+                        strs[j * 2 + k] += 7;
+                        }
+                    uid[j]>>=4;
+                }
+                }
+
+            for (len = 0; len < DESC_STR_MAX - 1 && strs[len]; ++len) {
+            desc_str[1 + len] = strs[len];
+        }
```
Comment (robert-hh) [inline ports/mimxrt/tusb_port.c:130]: DESC_STR_MAX is 20, but only the first 16 bytes of strs[] are set. So either change DESC_STR_MAX, or intialize [] to 0.

File: ports/mimxrt/tusb_port.c [hunk 2]
```diff
@@ -104,9 +109,31 @@ const uint16_t *tud_descriptor_string_cb(uint8_t index, uint16_t langid) {
         if (index >= sizeof(usbd_desc_str) / sizeof(usbd_desc_str[0])) {
             return NULL;
         }
+        if (index == USBD_STR_SERIAL)
+        {
+        uint8_t strs[DESC_STR_MAX];  
+        uint8_t uid[8];
+        mp_hal_get_unique_id(uid);
```
Comment (robert-hh) [inline ports/mimxrt/tusb_port.c:116]: may add here:
memset(strs, DESC_STR_MAX, 0);

**Review by robert-hh — COMMENTED** [review]
Just a few comments. b.t.w.: The commit message format must be:
<port>:  [A-Z].*\.
e.g.
mimxrt: Set the USB serial number.

You can edit the PR title in the browser, but that will not change the commit message.

**Comment by robert-hh** [comment]
Sorry for interfering. I just tested the PR and could not resist to make a slightly more compact version of some of the function tud_descriptor_string_cb():
```
const uint16_t *tud_descriptor_string_cb(uint8_t index, uint16_t langid) {
    #define DESC_STR_MAX (20)
    static uint16_t desc_str[DESC_STR_MAX];
    static const char hexchr[16] = "0123456789ABCDEF";

    uint8_t len;
    if (index == 0) {
        desc_str[1] = 0x0409; // supported language is English
        len = 1;
    } else {
        if (index >= sizeof(usbd_desc_str) / sizeof(usbd_desc_str[0])) {
            return NULL;
        }
        if (index == USBD_STR_SERIAL) {
            uint8_t uid[8];
            mp_hal_get_unique_id(uid);
            // store it as a hex string
            for (len = 0; len < 16; len += 2) {
                desc_str[1 + len] = hexchr[uid[len / 2] >> 4];
                desc_str[1 + len + 1] = hexchr[uid[len / 2] & 0x0f];
            }
        } else {
            const char *str = usbd_desc_str[index];
            for (len = 0; len < DESC_STR_MAX - 1 && str[len]; ++len) {
                desc_str[1 + len] = str[len];
            }
        }
    }

    // first byte is length (including header), second byte is string type
    desc_str[0] = (TUSB_DESC_STRING << 8) | (2 * len + 2);

    return desc_str;
}
```

File: ports/mimxrt/tusb_port.c [hunk 3]
```diff
@@ -25,12 +25,17 @@
  */
 
 #include "tusb.h"
+#include "mphalport.h"
 
 #ifndef MICROPY_HW_USB_VID
 #define MICROPY_HW_USB_VID (0xf055)
 #define MICROPY_HW_USB_PID (0x9802)
```
Comment (alphaFred) [inline ports/mimxrt/tusb_port.c:32]: If we are touching the tusb config then please add an `#ifndef MICROPY_HW_USB_PID` as well. In my point of view we should be able to configure both VID and PID individually.

Like:
```
#ifndef MICROPY_HW_USB_VID
#define MICROPY_HW_USB_VID (0xf055)
#endif
#ifndef MICROPY_HW_USB_PID
#define MICROPY_HW_USB_PID (0x9802)
#endif
```

Comment (d3v1c3nv11) [reply ports/mimxrt/tusb_port.c:32]: Olimex have the Christmas party right now, but you can look at my fork. I made some other changes

Comment (alphaFred) [reply ports/mimxrt/tusb_port.c:32]: Hey no worries - by all means enjoy your Christmas party. I just think that this suggestion is a worthy improvement which you should consider bringing in with your PR.

**Comment by robert-hh** [comment]
@dpgeorge This PR is part of PR #8162. The initial poster @d3v1c3nv11 seems to have left the party.

**Comment by dpgeorge** [comment]
Closing in favour of #8162.

**Pull request closed without merging**